### PR TITLE
plugin/snowflake: update gosnowflake to v1.6.1

### PR DESCRIPTION
**Pull request by jasonodonnell**
Snowflake plugin is crashing on 32bit OS's due to go snowflake being out of date. I've updated the dependencies, however, an azure had a signature change so it required a small change in Vault.

https://github.com/hashicorp/vault-plugin-database-snowflake/pull/8

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -36,6 +36,7 @@ require (
 	github.com/coreos/go-semver v0.2.0
 	github.com/coreos/go-systemd v0.0.0-20191104093116-d3cd4ed1dbcf
 	github.com/denisenkom/go-mssqldb v0.0.0-20200428022330-06a60b6afbbc
+	github.com/dgrijalva/jwt-go/v4 v4.0.0-preview1 // indirect
```
Comment (calvn) [inline go.mod:39]: If you run a `go mod tidy`, do the indirect deps go away?

Comment (jasonodonnell) [reply go.mod]: Indeed, sorry!

**Review by vinay-gopalan — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -10,7 +10,7 @@ require (
 	cloud.google.com/go v0.56.0
 	cloud.google.com/go/spanner v1.5.1
 	cloud.google.com/go/storage v1.6.0
-	github.com/Azure/azure-storage-blob-go v0.11.0
+	github.com/Azure/azure-storage-blob-go v0.13.0
```
Comment (calvn) [inline go.mod:13]: I a bit surprised that this got updated.

Comment (jasonodonnell) [reply go.mod:13]: @calvn gosnowflake uses it, is this a blocker?

Comment (calvn) [reply go.mod:13]: Understood. Nope, if snowflake requires the upgrade then that's fine.

**Review by calvn — APPROVED** [review]

**Pull request merged**